### ISSUE TITLE
Add Timedelta Support to JSON Reader with orient="table"

### ISSUE THREAD

**Issue by WillAyd** [pandas-dev/pandas#21140]
Right now trying to parse JSON using orient='table' will fail if the object contains Timedeltas intentionally:

https://github.com/pandas-dev/pandas/blob/bc37ea2e05019a89adaa48159b220483598d1898/pandas/io/json/table_schema.py#L310

Since #19065 was merged I think this should now be possible if anyone wants to take a look

**Comment by nprad** [comment]
I would like to work on this issue!

**Comment by fjdiod** [comment]
Hello, should pd.Timedelta be used for parsing here? I mean, use it somewhere inside parse_table_schema?

**Comment by WillAyd** [comment]
Yes IIRC correctly the limiting factor previously was that timedeltas are represented as ISO 8601 durations in the JSON table Schema, but when this was first built the Timedelta constructor would not parse that duration format as an argument. 

Since the PR in the original post that is no longer an issue so this should be possible to do now

**Comment by fjdiod** [comment]
Now, if parse durations with Timedelta constructor in parse_table_schema, they get coerced into int64 after, in FrameParser from json.py(it seems that it tries coerce floats without deciaml digits). Is it expected behavior? Maybe their dtype should be changed from timedelta64 to timedelta64[ns]?
